### ISSUE TITLE
sphinx documentation with `methodtools.lru_cache`

### ISSUE THREAD

**Issue by jbloom** [youknowone/methodtools#1]
When I decorate a method with `methodtools.lru_cache` and then build sphinx documentation, the docstring shows as `<methodtools._LruCacheWire object>` rather than the documentation for the method in its docstring.

I think this type of problem is addressed [here](https://stackoverflow.com/a/56411967), but I can't figure out how to modify `methodtools` to fix the problem.

**Comment by youknowone** [comment]
Thanks for reporting missing feature.
Because it is implemented out of methodtools project, I made a PR to wirerope.
https://github.com/youknowone/wirerope/pull/5

Unfortunately, I made working version for methods but not about properties yet.

**Comment by youknowone** [comment]
I released 0.1.2 with this document fix: https://pypi.org/project/methodtools/0.1.2/
Will you test it please?

**Comment by jbloom** [comment]
This did not seem to fix the problem for me.

**Comment by youknowone** [comment]
I wanted to test this problem with actual example of sphinx but didn't have enough time to do that.
Do you know any small and simple sphinx example proejct?

**Comment by lisphilar** [comment]
Dear @youknowone ,
Thank you for your helpful package.
I started to use `methodtools` in my project and found the same issue.
https://lisphilar.github.io/covid19-sir/covsirphy.cleaning.html#covsirphy.cleaning.term.Term.num2str

Do we need to create a new decorator to show the docstring?
Please help me out.

**Comment by youknowone** [comment]
Sorry for late fix. I want to fix this problem but almost forgot about it for a while. I think wrapping once more will be a simple workaround for simple use case, if you don't use `cache_info()`

![image](https://user-images.githubusercontent.com/69878/88474194-35d1b500-cf5f-11ea-9b48-6ec348bd1134.png)

**Comment by lisphilar** [comment]
Dear @youknowone ,
Thank you for your prompt reply!
I tried the following code.

```Python
from methodtools import lru_cache

def show_info(func):
    def wrapper(*args, **kwargs):
        return func.cache_info(*args, **kwargs)
    return wrapper


class ClassA(object):
    @show_info
    @lru_cache(maxsize=None)
    def method_a(self, arg1):
        """
        This may be a method to do something.
        """
        pass
```
In sphinx document, `<methodtools._LruCacheWire object>` was successfully removed!
However, it shows `method_a(**kwargs)` even when arguments (`arg1`) exist. Docsting was not shown.
Sorry for asking basic question of decorator, but please tell me how to improve it.

**Comment by youknowone** [comment]
I don't think you need `cache_info` here. Check this way: https://docs.python.org/3/library/functools.html#functools.wraps

**Comment by lisphilar** [comment]
Dear @youknowone ,
I'm sorry, but I couldn't find the solution in my project.
Please clarify the solution with the example codes I used in the previous comment.

**Comment by youknowone** [comment]
It seems sphinx only support actual functions

**Comment by lisphilar** [comment]
Thank you for your help.
